### PR TITLE
CRAYSAT-1343: Add SAT support

### PR DESCRIPTION
**Pull request by ekamin-hpe**
## Original Commit Message

```
This commit modifies the build scripts so that they source a config file
which contains product-specific configuration values, so that the scripts
are not hard-coded to CSM. To build documentation for SAT, give
a positional argument to the build script, e.g. `bin/build.sh sat`.
To maintain backward compatibility, running the scripts without any
arguments will continue to build the CSM docs.

This script adds two config files in conf/
* csm.sh (default)
* sat.sh

A couple other files are modified for SAT support:
* docker-compose files are split up, adding sat-specific compose files
  in bin/compose/sat/. These files are referred to in sat.sh.
* A new Hugo config file, config.sat.toml is added at the top of this
  repo. It defines the different "languages" which are actually used for
  switching between versions.
* The Hugo config files get a new parameter called shortTitle. This is
  referred to in themes/hugo-theme-learn/layouts/partials/logo.html
  rather than hard-coding the doc title in the theme.
* README.md is modified to provide some short instructions on how to
  build the SAT docs.
* Modify convert-docs-to-hugo.sh so that the value "index.md" is
  configurable, so that a repo that uses README.md files as the index
  can be built in the same way.

Finally, add sat_docs_build.log and sat-docs to .gitignore.

Test Description:
* I ran the following scripts with no arguments to make sure CSM
  docs were still building as expected:
  * bin/build.sh
  * bin/dev.sh
  * bin/test.sh
  * bin/push.sh (For this test, I commented out the line that actually
    performs the push).
* Next, I ran the following to test building SAT docs:
  * bin/build.sh sat
  * bin/dev.sh sat
  * bin/test.sh sat
  * bin/push.sh sat
```

## Summary and Scope

This change adds support for building SAT docs using the csm-docs-html-builder code. It does this by adding support for a configuration file, e.g. `conf/csm.sh` or `conf/sat.sh`. The default behavior is to build CSM docs, so this change is fully backward compatible.

## Issues and Related PRs

* CRAYSAT-1341

### Tested on:

  * Local development environment

### Test description:

See commit message

## Risks and Mitigations

N/A

## Pull Request Checklist

- [X] Version number(s) incremented, if applicable
- [X] Copyrights updated
- [X] License file intact
- [X] Target branch correct
- [X] CHANGELOG.md updated (N/A)
- [X] Testing is appropriate and complete, if applicable
- [X] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable (N/A)

### CODE REVIEW

**Comment by mtupitsyn** [comment]
Hi Eli,

Overall this looks good to me. Couple of questions / minor tweaks:
1. We run CSM HTML generator nightly: https://github.com/Cray-HPE/csm-docs-html-builder/blob/master/.github/workflows/build-docs.yaml#L30.  I assume SAT will also need nightly workflow? Better add it to this PR (just shift timing a little bit).
2. We have a license check as mandatory exercise on all PR's (it did not yet ran against your change, as workflows in general, including organizational workflows are disabled on a fork). It is going to fail when this gets merged, as license must be present in each file changed/added within PR. You can run it locally using this command:
\
    docker run -it --rm -v $(pwd):/github/workspace artifactory.algol60.net/csm-docker/stable/license-checker .
\
Accordingly, you can let it fix all license headers for you:
\
    docker run -it --rm -v $(pwd):/github/workspace artifactory.algol60.net/csm-docker/stable/license-checker --fix .
\
I don't exactly recall if license checker allows having comments before license header. Better always have license coming right after shebang line (for sh files). E.g. in `conf/csm.sh` file, please move lines 1-3 explaining purpose of this file down to after license header (you can add shebang line as first line though).

File: conf/sat.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+# A list of releases that should be built. Each entry should be X.Y where the
+# branch release/X.Y exists in the docs repo.
+export BRANCHES=("2.3" "2.2" "2.1")
+
+# The documentation repository remote URL. It must be possible to run
+# 'git clone $DOCS_REPO_REMOTE_URL'.
+export DOCS_REPO_REMOTE_URL="git@github.hpe.com:eli-kamin/docs-sat.git"
```
Comment (mtupitsyn) [inline conf/sat.sh:34]: Any plans to move docs-sat repo under Cray-HPE umbrella?

**Review by mtupitsyn — APPROVED** [review]
Looks good, but needs a license check fix.

File: conf/sat.sh (hunk 1 above)
Comment (ekamin-hpe) [reply conf/sat.sh:34]: See https://github.com/Cray-HPE/csm-docs-html-builder/pull/6#issuecomment-1118744741

**Review by mtupitsyn — APPROVED** [review]

**Comment by mtupitsyn** [comment]
> I pushed [3cd2133](https://github.com/Cray-HPE/csm-docs-html-builder/commit/3cd2133c495f8d9cac9b39273d5e468190582e70) addressing the license check issue. I also found a couple extra issues which I addressed in [923b2bf](https://github.com/Cray-HPE/csm-docs-html-builder/commit/923b2bfecdf236fd19d330ed3f093c3e677bbdfc). I'll be happy to squash the commits when we're ready to merge.

Sounds good! Let me know when (if) it's ready to merge, I'll do a squash merge.

**Comment by mtupitsyn** [comment]
No problem, I’ve re-approved. I am not sure if you can merge - if not, let me know, I can merge.

**Comment by ekamin-hpe** [comment]
> I am not sure if you can merge - if not, let me know, I can merge.

I can't. Please merge at your convenience, thank you!

**Comment by mtupitsyn** [comment]
Done!

**Pull request merged**